app/services/channel_queues/response_retriever.rb

```diff
@@ -27,7 +27,13 @@ def self.retrieve(original_request_text, channel_id, channel_name, user_id, user
           return json_result
         end
 
-        ChannelQueueMembership.create!(user: user, channel_queue: channel_queue)
+        new_membership = ChannelQueueMembership.create!(user: user, channel_queue: channel_queue)
```

Comment:
Do you think it'll be better to block ChannelQueueMembership creation during inactive hours? The current logic creates the invalid membership then relies on another method call to delete it. By blocking the creation we won't have to scope for active/inactive either since they simply wouldn't exist.

Reply:
yeah i can change that

Reply:
done https://github.com/choruk/foos_portal/pull/43/commits/0b51851c08d0ead4293f2fddcb0235ad60503741